### PR TITLE
Fix hang with jobs=1 and add offline Ethereum key detection

### PR DESCRIPTION
**Pull request by mickgmdb**
This pull request introduces deterministic, network-free Ethereum key and BIP-39 mnemonic detection and validation, along with improvements to validation outcomes and internal security. The changes include new rules for Ethereum private keys and mnemonics, a reusable BIP-39 checksum filter, and explicit reporting of local cryptographic validation results. The update also hardens validation caching and panic handling to prevent secret exposure and fixes a hang in remote Git URL scans.

**Ethereum and BIP-39 Key Detection and Validation:**

- Added new rules for detecting Ethereum private keys, public keys, and BIP-39 seed phrases, including support for Ethereum-specific mnemonics. These rules leverage local cryptographic validation to determine if detected material is structurally valid. (`crates/kingfisher-rules/data/rules/ethereum.yml`, `crates/kingfisher-rules/data/rules/bip39.yml`) [[1]](diffhunk://#diff-e4c55c5029f0199f94eaff395e3e86e9a800c4ba345600206fefd9cdb182bc9cR1-R155) [[2]](diffhunk://#diff-3b66c75c20d37493b468d18e080fffb250c97da679ffff698c4011d51a25a17eR1-R53)
- Introduced a reusable `bip39_valid` Liquid filter for checksum validation of BIP-39 mnemonics, with comprehensive tests for standard and edge cases. (`crates/kingfisher-rules/src/liquid_filters.rs`) [[1]](diffhunk://#diff-26bb9d129d373a554cb90fb91ce6e2494bb2b2fbfa57aece10c3ec5011b5cbc4R528-R547) [[2]](diffhunk://#diff-26bb9d129d373a554cb90fb91ce6e2494bb2b2fbfa57aece10c3ec5011b5cbc4R1410-R1437)

**Validation Outcome and Reporting Improvements:**

- Extended `ValidationOutcome` to explicitly represent locally derived (network-free) cryptographic validation results and invalid material, with corresponding serialization and display names. (`crates/kingfisher-core/src/validation.rs`) [[1]](diffhunk://#diff-b57b208571f4e9619aaf846f7b9d120a6228650a68cab63aab08e90e3f955351L32-R36) [[2]](diffhunk://#diff-b57b208571f4e9619aaf846f7b9d120a6228650a68cab63aab08e90e3f955351L52-R65) [[3]](diffhunk://#diff-b57b208571f4e9619aaf846f7b9d120a6228650a68cab63aab08e90e3f955351R103-R104) [[4]](diffhunk://#diff-b57b208571f4e9619aaf846f7b9d120a6228650a68cab63aab08e90e3f955351R134-R141)
- Updated actionable validation logic to include locally derived outcomes, improving reporting accuracy. (`crates/kingfisher-core/src/validation.rs`)

**Validation Infrastructure and Security:**

- Added support for deterministic Ethereum key validation as a feature-gated dependency, including new workspace dependencies and feature flags. (`Cargo.toml`, `crates/kingfisher-scanner/Cargo.toml`, `crates/kingfisher-scanner/src/lib.rs`) [[1]](diffhunk://#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R21-R22) [[2]](diffhunk://#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R38) [[3]](diffhunk://#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R49-R54) [[4]](diffhunk://#diff-1bd728aadd743b9b405d529d969c3d912e8ac092110e6c61749f651d3855187bR40) [[5]](diffhunk://#diff-1bd728aadd743b9b405d529d969c3d912e8ac092110e6c61749f651d3855187bR50) [[6]](diffhunk://#diff-b7b33f9d38cbcd5ecd1b425ec0ba76e3ba68cee7e6304f75e7e22dbf48359355R46-R54) [[7]](diffhunk://#diff-b7b33f9d38cbcd5ecd1b425ec0ba76e3ba68cee7e6304f75e7e22dbf48359355R124) [[8]](diffhunk://#diff-b7b33f9d38cbcd5ecd1b425ec0ba76e3ba68cee7e6304f75e7e22dbf48359355R179-R184) [[9]](diffhunk://#diff-e8d911650e9a02bb08b9372770d555c10d80483b6bdf0bf3990db041b50b9a23R51)
- Hardened validation caching and panic handling to avoid secret exposure. (`CHANGELOG.md`)

**Other Fixes and Updates:**

- Fixed an issue where remote Git URL scans could hang when running with `--jobs 1`. (`CHANGELOG.md`)
- Bumped version to `v1.112.0` and updated changelog to reflect these changes. (`Cargo.toml`, `CHANGELOG.md`) [[1]](diffhunk://#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R49-R54) [[2]](diffhunk://#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR5-R9)

### CODE REVIEW

File: crates/kingfisher-rules/src/liquid_filters.rs [hunk 1]
```diff
@@ -17,6 +18,7 @@ use time::{
     format_description::well_known::{Iso8601, Rfc2822},
 };
 use uuid::Uuid;
+use zeroize::Zeroizing;
```
[the remark this answers is not in the document]
Comment (mickgmdb) [reply crates/kingfisher-rules/src/liquid_filters.rs:21]: /ar This is an intentional defensive use of `zeroize::Zeroizing`. The filter temporarily holds the complete BIP-39 mnemonic while normalizing and checksum-validating it, then wipes the temporary on drop. It does not sign, encrypt, or transmit the mnemonic; no code change is needed.

File: crates/kingfisher-scanner/src/validation/ethereum.rs [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    }
+}
+
+fn strip_hex_prefix(value: &str) -> &str {
+    value.strip_prefix("0x").or_else(|| value.strip_prefix("0X")).unwrap_or(value)
+}
+
+fn address_from_public_key(key: &VerifyingKey) -> String {
+    let encoded = key.to_encoded_point(false);
+    let digest = Keccak256::digest(&encoded.as_bytes()[1..]);
```
[the remark this answers is not in the document]
Comment (mickgmdb) [reply crates/kingfisher-scanner/src/validation/ethereum.rs:175]: /ar Keccak-256 is intentionally required for Ethereum address derivation. The input is the uncompressed secp256k1 public key without its 0x04 prefix, and the final 20 bytes form the address. There is no secret-key, nonce, encryption, or MAC operation. This is protocol-defined and covered by known-address tests.

File: crates/kingfisher-scanner/src/validation/ethereum.rs [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    checksum_address(&digest[12..])
+}
+
+fn checksum_address(address: &[u8]) -> String {
+    debug_assert_eq!(address.len(), 20);
+    let lower = hex::encode(address);
+    let hash = Keccak256::digest(lower.as_bytes());
```
[the remark this answers is not in the document]
Comment (mickgmdb) [reply crates/kingfisher-scanner/src/validation/ethereum.rs:182]: /ar Keccak-256 is intentionally required for Ethereum address derivation. The input is the uncompressed secp256k1 public key without its 0x04 prefix, and the final 20 bytes form the address. There is no secret-key, nonce, encryption, or MAC operation. This is protocol-defined and covered by known-address tests.

File: crates/kingfisher-scanner/src/validation/ethereum.rs [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+use bip32::{
+    DerivationPath, XPrv,
+    secp256k1::ecdsa::{SigningKey, VerifyingKey},
+};
+use bip39::{Language, Mnemonic};
+use kingfisher_core::ValidationOutcome;
+use kingfisher_rules::EthereumValidation;
+use serde::{Deserialize, Serialize};
+use sha3::{Digest, Keccak256};
+use zeroize::{Zeroize, Zeroizing};
```
[the remark this answers is not in the document]
Comment (mickgmdb) [reply crates/kingfisher-scanner/src/validation/ethereum.rs:12]: /ar This is an intentional crypto-validation implementation. The `validation-ethereum` feature uses `zeroize` to wipe mnemonic normalization, BIP-39 seed material, and private-key byte buffers. Validation is deterministic and network-free; no remediation is needed.
